ddtrace/span.py

```diff
@@ -168,6 +169,10 @@ def __init__(
 
         # sampling
         self.sampled = True  # type: bool
+        # 0 = no sampling has run
```

Comment:
The code comment essentially explains what my interpretation of the RFC is. We need a way to figure out where `sample` has run, to figure out if we should run the `sample` again. For example, the `sample` call in propagation/http.py should override the sampling decision if `sample` was called in  processor/__init__ (before encoding).  Neither of those sample calls should override manually setting sampling_priority in code on the span.

Reply:
I need to write some tests for this overriding behavior.